java/client/src/main/java/glide/api/models/commands/RestoreOptions.java

```diff
@@ -0,0 +1,96 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands;
+
+import glide.api.commands.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import lombok.*;
+
+/**
+ * Optional arguments to {@link GenericBaseCommands#restore(byte[], long, byte[], RestoreOptions)}
+ *
+ * @see <a href="https://valkey.io/commands/restore/">valkey.io</a>
+ */
+@Getter
+@Builder
+public final class RestoreOptions {
+    /** <code>REPLACE</code> subcommand string to replace existing key */
+    public static final String REPLACE_REDIS_API = "REPLACE";
+
+    /**
+     * <code>ABSTTL</code> subcommand string to represent absolute timestamp (in milliseconds) for TTL
+     */
+    public static final String ABSTTL_REDIS_API = "ABSTTL";
+
+    /** <code>IDELTIME</code> subcommand string to set Object Idletime */
+    public static final String IDLETIME_REDIS_API = "IDLETIME";
+
+    /** <code>FREQ</code> subcommand string to set Object Frequency */
+    public static final String FREQ_REDIS_API = "FREQ";
+
+    /** When `true`, it represents <code>REPLACE</code> keyword has been used */
+    private final boolean hasReplace;
```

Comment:
You could create a replace() and absttl() method that will set (true) automatically which looks a little better than hasReplace(true).  
example: https://github.com/Bit-Quill/glide-for-redis/blob/main/java/client/src/main/java/glide/api/models/commands/stream/StreamGroupOptions.java#L32-L36